src/pages/DataAccessRequestApplication.js

```diff
@@ -213,45 +205,36 @@ class DataAccessRequestApplication extends Component {
-    formData.orcid = rpProperties.orcid !== undefined ? rpProperties.orcid : '';
-    formData.institution = isNil(researcher)  || isNil(researcher.institutionId)? "" : (await Institution.getById(researcher.institutionId)).name;
-    formData.department = rpProperties.department != null ? rpProperties.department : '';
-    formData.division = rpProperties.division != null ? rpProperties.division : '';
-    formData.address1 = rpProperties.address1 != null ? rpProperties.address1 : '';
-    formData.address2 = rpProperties.address2 != null ? rpProperties.address2 : '';
-    formData.city = rpProperties.city != null ? rpProperties.city : '';
-    formData.zipCode = rpProperties.zipCode != null ? rpProperties.zipCode : '';
-    formData.country = rpProperties.country != null ? rpProperties.country : '';
-    formData.state = rpProperties.state != null ? rpProperties.state : '';
-    formData.piName = rpProperties.piName !== null ? rpProperties.piName : '';
-    formData.academicEmail = rpProperties.academicEmail != null ? rpProperties.academicEmail : '';
-    formData.piEmail = rpProperties.piEmail != null ? rpProperties.piEmail : '';
-    formData.isThePi = rpProperties.isThePI !== undefined ? rpProperties.isThePI : '';
-    formData.havePi = rpProperties.havePI !== undefined ? rpProperties.havePI : '';
-    formData.pubmedId = rpProperties.pubmedID !== undefined ? rpProperties.pubmedID : '';
-    formData.scientificUrl = rpProperties.scientificURL !== undefined ? rpProperties.scientificURL : '';
-    formData.userId = Storage.getCurrentUser().dacUserId;
+    formData.investigator = rpProperties.piName;
+
+    formData.linkedIn = rpProperties.linkedIn;
+    formData.researcherGate = rpProperties.researcherGate;
+    formData.orcid = rpProperties.orcid;
+    formData.institution = isNil(researcher)  || isNil(researcher.institution)? "" : researcher.institution.name;
+    formData.department = rpProperties.department;
+    formData.division = rpProperties.division;
+    formData.address1 = rpProperties.address1;
+    formData.address2 = rpProperties.address2;
+    formData.city = rpProperties.city;
+    formData.zipCode = rpProperties.zipCode;
+    formData.country = rpProperties.country;
+    formData.state = rpProperties.state;
+    formData.piName = rpProperties.piName;
+    formData.academicEmail = rpProperties.academicEmail;
+    formData.piEmail = rpProperties.piEmail;
+    formData.isThePi = rpProperties.isThePI;
+    formData.havePi = rpProperties.havePI;
+    formData.pubmedId = rpProperties.pubmedID;
+    formData.scientificUrl = rpProperties.scientificURL;
+    formData.userId = researcher.dacUserId;
 
     let completed = false;
     if (!fp.isNil(formData.darCode)) {
       completed = '';
-    } else if (rpProperties.completed !== undefined) {
-      completed = JSON.parse(rpProperties.completed);
+    } else if (rpProperties.completed !== '') {
```

Comment:
`rpProperties.completed` can still be nil or undefined if `getPropertyValuesFromUser` wasn't able to find a matching property. In fact, I think that's true of all of the `rpProperties` fields, so all of the simplifications above should be reverted.

Reply:
if the property is null then the helper returns the empty string.  utils.getPropertyValuesFromUser calls utils.findPropertyValue which does the null checking and replacing

Reply:
Ah, thank you, yes, that works.